src/module/apps/acc_diff/plugins/brawler.ts

```diff
@@ -0,0 +1,68 @@
+import { AccDiffHudData, AccDiffHudTarget } from "../data";
+import { AccDiffHudCheckboxPluginData, AccDiffHudPluginCodec } from "./plugin";
+import { enclass } from "../../serde";
+import { AccDiffWindowType, WeaponType } from "../../../enums";
+import { slugify } from "../../../util/lid";
+import { AbstractTalent } from "./abstractTalents";
+
+//Automated with a caveat, see talent()
+export default class Brawler_1 extends AbstractTalent implements AccDiffHudCheckboxPluginData {
+  //Shared type requirements
+  //slugify here to make sure the slug is same across this plugin and TalentWindow.svelte
+  //Alternatively could use lid and rank_num
+  static slug: string = slugify("Hold and Lock", "-");
+  slug: string = slugify("Hold and Lock", "-");
+  static kind: "hase" | "attack" = "attack";
+  kind: "hase" | "attack" = "attack";
+  lid: string = "t_brawler";
+  talentRank: number = 1;
+  humanLabel: string = "Hold and Lock";
+  quickReference: string = "+1";
+  tooltip: string = "You gain +1 Accuracy on all melee attacks against targets YOU are Grappling.";
+
+  //AccDiffHudPlugin requirements
+  // the codec lets us know how to persist whatever data you need for rerolls
+  static get codec(): AccDiffHudPluginCodec<Brawler_1, unknown, unknown> {
+    return enclass(this.schemaCodec, Brawler_1);
+  }
+
+  //We care about individual targets, so we do both
+  static perUnknownTarget(): Brawler_1 {
+    let ret = new Brawler_1();
+    return ret;
+  }
+  static perTarget(item: Token): Brawler_1 {
+    return Brawler_1.perUnknownTarget();
+  }
+
+  //The unique logic of the talent
+  talent(data: AccDiffHudData, target?: AccDiffHudTarget) {
+    if (data.targets.length > 0 && target === undefined) return;
```

Comment:
As above, I'd prefer `!target` instead of `target === undefined`. It's shorter, and catches all falsey cases. There's a bunch of other places this approach can be applied too.